### PR TITLE
[ISSUE-14] Add sha256 and sha384 support for kms

### PR DESCRIPTION
**Pull request by ooq**
Addressing #14 .

### CODE REVIEW

File: signer/kms/signer.go [hunk 1]
```diff
@@ -45,15 +49,43 @@ func (s *Signer) Sign(checkName, body string) (string, string, error) {
...
+	default:
+		return "", "", fmt.Errorf("Unsupported digest algorithm %v", key.Algo)
+	}
+
+	if _, err := digest.Write([]byte(body)); err != nil {
+		return "", "", err
+	}
+
+	var d kms_pb.Digest
+	switch key.Algo {
```
Comment (Nilay-Shah) [inline signer/kms/signer.go:64]: why not group this switch statement with the 1 above so there's only 1 switch statement?

**Review by Nilay-Shah — COMMENTED** [review]
1 question. overall looks good though!

File: signer/kms/signer.go (hunk 1 above)
Comment (catherinejones) [reply signer/kms/signer.go:64]: My guess is performance! If the algorithm is unsupported, don't bother writing anything. Might I suggest moving this switch statement into the `NewSigner` method, that way we don't create the signer if it'll be misconfigured?

**Review by catherinejones — APPROVED** [review]
The only change I'd request would be to move the switch statement into the `NewSigner` method, but other than that this looks good to me!

**Review by rxbchen — APPROVED** [review]
💯 to everything mentioned above, otherwise looks good!

File: signer/kms/signer.go (hunk 1 above)
Comment (ooq) [reply signer/kms/signer.go:64]: Done.

**Review by Nilay-Shah — APPROVED** [review]

**Pull request merged**